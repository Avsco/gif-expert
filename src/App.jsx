function App() {
  return (
    <div>
      <h1>GifExpertApp</h1>
    </div>
  );
}

export default App;
