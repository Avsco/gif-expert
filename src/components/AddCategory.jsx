import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({ onAddCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      onAddCategories(inputValue);
      setInputValue('');
    }
  };

  return (
    <form style={{ display: 'flex', gap: 16 }} onSubmit={handleSubmit} aria-label='form'>
      <input placeholder='Buscar gifs' value={inputValue} onChange={handleInputChange} />
      <button type='submit'>Agregar</button>
    </form>
  );
};

AddCategory.propTypes = {
  onAddCategories: PropTypes.func.isRequired,
};
