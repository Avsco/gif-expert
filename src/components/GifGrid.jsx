import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';
import { GifItem } from './GifItem';

export const GifGrid = ({ category }) => {
  const [gifs, setGifs] = useState([]);

  useEffect(() => {
    getGifs(category).then((gifs) => {
      setGifs(gifs);
    });
  }, [category]);

  return (
    <div>
      <h3>{category}</h3>
      <div className='card-grid'>
        {gifs.map(({ id, url, title }) => (
          <GifItem key={id} id={id} url={url} title={title} />
        ))}
      </div>
    </div>
  );
};
