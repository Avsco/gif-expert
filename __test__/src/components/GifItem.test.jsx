import { render } from '@testing-library/react';
import { GifItem } from '../../../src/components';

describe('Test in <GifItem/>', () => {
  const id = '1';
  const url = 'https://media1.giphy.com/media/xTcnKDqEQqQqQ/200.gif';
  const title = 'Test title';

  test('Evaluar snapshot', () => {
    const { asFragment } = render(<GifItem id={id} url={url} title={title} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Mostrar la imagen con el URL indicado y el alt indicado', () => {
    const { container } = render(<GifItem id={id} url={url} title={title} />);

    expect(container.querySelector('img').getAttribute('src')).toBe(url);
    expect(container.querySelector('img').getAttribute('alt')).toBe(title);
  });

  test('Debe de mostrar el titulo del componente', () => {
    const { getByText } = render(<GifItem id={id} url={url} title={title} />);

    expect(getByText(title)).toBeTruthy();
  });
});
