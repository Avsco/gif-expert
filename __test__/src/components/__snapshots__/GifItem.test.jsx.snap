// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem/> Evaluar snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <img
      alt="Test title"
      src="https://media1.giphy.com/media/xTcnKDqEQqQqQ/200.gif"
    />
    <p>
      Test title
    </p>
  </div>
</DocumentFragment>
`;
